process/block/baseProcess.go

```diff
@@ -988,6 +988,16 @@ func (bp *baseProcessor) saveBody(body *block.Body) {
 		log.Trace("saveBody.Put -> MiniBlockUnit", "time", time.Since(startTime))
 	}
 
+	marshalizedReceiptsHashes, errNotCritical := bp.txCoordinator.CreateMarshalizedReceipts()
+	if errNotCritical != nil {
+		log.Warn("saveBody.CreateMarshalizedReceipts", "error", errNotCritical.Error())
+	}
+
+	errNotCritical = bp.store.Put(dataRetriever.ReceiptsUnit, header.GetReceiptsHash(), marshalizedReceiptsHashes)
```

Comment:
if there was an error previously you are storing nil, I think you should skip instead

Reply:
Yep, also it should be avoided if len(shardHdr.GetReceiptsHash()) or len(marshalizedReceiptsHashes) is 0 I think